### PR TITLE
Bump Updates

### PR DESCRIPTION
**Pull request by bitPogo**
### :tophat: What is the goal?
AGP 7.2.2
Kotlin 1.7.10
mockito 4.7.0
jsonassert 1.5.1

### :thinking: DOD Checklist

- [x] My code follows the code style and naming conventions of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the documentation.
- [x] I have updated the changelog accordingly.

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -36,11 +36,11 @@ dependencies {
...
+    implementation("com.github.ben-manes:gradle-versions-plugin:0.42.0")
     // download-scripts.gradle.kts
-    implementation("de.undercouch:gradle-download-task:4.1.2")
+    implementation("de.undercouch:gradle-download-task:5.1.1")
     // publishing.gradle.kts
-    implementation("org.eclipse.jgit:org.eclipse.jgit:5.11.0.202103091610-r")
+    implementation("org.eclipse.jgit:org.eclipse.jgit:6.2.0.202206071550-r")
     // versioning.gradle.kts
-    implementation("care.data4life.gradle.gitversion:gradle-git-version:0.12.4-d4l")
+    implementation("com.palantir.gradle.gitversion:gradle-git-version:0.15.0")
```
Comment (wmontwe) [inline buildSrc/build.gradle.kts:45]: Is this version working with nested git projects and submodules? That was the reason to have a forked version.

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -43,6 +43,11 @@ toc::[]
 
 === Bumped
 
+* Gradle 7.4.2 -> 7.5.1
+* Android Gradle Plugin 7.1.3 -> 7.2.2
+* _**BREAKING**_ Kotlin 1.6.10 -> 1.7.0
```
Comment (wmontwe) [inline CHANGELOG.adoc:48]: 1.7.0 -> 1.7.10

**Review by wmontwe — CHANGES_REQUESTED** [review]
Have a question for git version plugin and a small doc update

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (bitPogo) [reply buildSrc/build.gradle.kts:45]: As far as I am aware of it does - I used it for my personal project and did not run into any issues this far.

File: CHANGELOG.adoc (hunk 2 above)
Comment (bitPogo) [reply CHANGELOG.adoc]: Done

**Review by wmontwe — APPROVED** [review]
👍

**Pull request merged**